h2/src/main/org/h2/command/Parser.java

```diff
@@ -10209,11 +10209,20 @@ private CreateLinkedTable parseCreateLinkedTable(boolean temp,
         }
         command.setOriginalTable(originalTable);
         read(CLOSE_PAREN);
-        if (readIf("EMIT")) {
+        if(readIf("FETCH_SIZE")){      
+            command.setFetchSize(readExpression().optimize(session));            
+        }
+        else if (readIf("EMIT")) {
             read("UPDATES");
             command.setEmitUpdates(true);
+            if(readIf("FETCH_SIZE")){      
+            command.setFetchSize(readExpression().optimize(session));            
+            }
         } else if (readIf("READONLY")) {
             command.setReadOnly(true);
+            if(readIf("FETCH_SIZE")){      
+            command.setFetchSize(readExpression().optimize(session));            
+            }
         }
```

Comment:
It is overcomplicated, just add a new separate `if (readIf("FETCH_SIZE")) {` below existing conditions.

Reply:
Thanks for the comment but I don't understand.

Reply:
```Java
        // existing code
        if (readIf("EMIT")) {
            read("UPDATES");
            command.setEmitUpdates(true);
        } else if (readIf("READONLY")) {
            command.setReadOnly(true);
        }
        // new code
        if (readIf("FETCH_SIZE")) {
            .....
        }
```

Reply:
Sorry it's ok